### PR TITLE
Return reload status from http endpoint (#1152)

### PR DESCRIPTION
**Pull request by stuartnelson3**

### CODE REVIEW

File: ui/web.go [hunk 1]
```diff
@@ -72,8 +73,16 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
...
 
 	r.Post("/-/reload", func(w http.ResponseWriter, req *http.Request) {
-		w.Write([]byte("Reloading configuration file..."))
-		reloadCh <- struct{}{}
+		errc := make(chan error)
+		defer close(errc)
+
+		reloadCh <- errc
+		if err := <-errc; err != nil {
+			http.Error(w, fmt.Sprintf("config file reload failed: %v", err), http.StatusInternalServerError)
```
Comment (stuartnelson3) [inline ui/web.go]: not sure if it's necessary to provide the error via http, or it's enough to just say the reload failed. Open to opinions.

File: ui/web.go [hunk 2]
```diff
@@ -72,8 +73,16 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
 	))
 
 	r.Post("/-/reload", func(w http.ResponseWriter, req *http.Request) {
-		w.Write([]byte("Reloading configuration file..."))
-		reloadCh <- struct{}{}
+		errc := make(chan error)
+		defer close(errc)
+
+		reloadCh <- errc
```
Comment (brancz) [inline ui/web.go]: I don't think passing around channels like this is appropriate here, it's only ever going to be a single error received from `errc`.

File: ui/web.go [hunk 3]
```diff
@@ -72,8 +73,16 @@ func Register(r *route.Router, reloadCh chan<- struct{}, logger log.Logger) {
...
+		if err := <-errc; err != nil {
+			http.Error(w, fmt.Sprintf("config file reload failed: %v", err), http.StatusInternalServerError)
+			return
+		}
+
+		w.Write([]byte("config file reloaded"))
```
Comment (brancz) [inline ui/web.go]: An empty 200 response is probably just as good. What does the Prometheus server do? I would out of consistency do the same.

**Comment by brian-brazil** [comment]
I wouldn't worry about exposing information here, all sorts of information is available via the web interface already including the ability to create and delete alerts. The only thing that'd be a problem would be secrets, and it's a bug to expose those in logs too.

I think this feature is a good idea.

File: ui/web.go (hunk 2 above)
Comment (stuartnelson3) [reply ui/web.go]: Coincidentally, it seems prometheus is using this pattern: https://github.com/prometheus/prometheus/blob/master/web/web.go#L638-L644

File: ui/web.go (hunk 3 above)
Comment (stuartnelson3) [reply ui/web.go]: Same link from above: https://github.com/prometheus/prometheus/blob/master/web/web.go#L638-L644

Looks like we're only writing something on failure

File: ui/web.go (hunk 2 above)
Comment (brancz) [reply ui/web.go]: I've been trying to think of a solution, but can't off the top of my head, so I think it's fine to go with it then.

**Comment by brancz** [comment]
lgtm on green :+1:

**Pull request merged**